src/sentry/static/sentry/app/actionCreators/savedSearches.jsx

```diff
@@ -58,14 +59,46 @@ export function saveRecentSearch(api, orgId, type, query) {
  */
 export function fetchRecentSearches(api, orgId, type, query) {
   const url = getRecentSearchUrl(orgId);
-  const promise = api
-    .requestPromise(url, {
-      query: {
-        query,
-        type,
-      },
-    })
-    .catch(handleXhrErrorResponse('Unable to fetch recent searches'));
+  const promise = api.requestPromise(url, {
+    query: {
+      query,
+      type,
+      limit: MAX_RECENT_SEARCHES,
+    },
+  });
+
+  promise.catch(handleXhrErrorResponse('Unable to fetch recent searches'));
+
+  return promise;
+}
+
+const getPinSearchUrl = orgId => `/organizations/${orgId}/pinned-searches/`;
+
+export function pinSearch(api, orgId, query, type) {
+  const url = getPinSearchUrl(orgId);
+  const promise = api.requestPromise(url, {
+    method: 'PUT',
+    data: {
+      query,
+      type,
+    },
+  });
+
+  promise.catch(handleXhrErrorResponse('Unable to pin search'));
+
+  return promise;
+}
+
+export function unpinSearch(api, orgId, type) {
+  const url = getPinSearchUrl(orgId);
+  const promise = api.requestPromise(url, {
+    method: 'DELETE',
+    data: {
+      type,
+    },
+  });
+
+  promise.catch(handleXhrErrorResponse('Unable to un-pin search'));
```

Comment:
Do we want to chain this so we don't have the uncaught exception? Or is there a plan to use the returned promise value in future (it's not currently used)?

Reply:
Yeah this is intentional so we can handle the errors in a user friendly manner downstream